Tests/State/WaitingNetworkStateSpecs.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  WaitingNetworkStateSpecs.swift
+//  ModernAVPlayer_Tests
+//
+//  Created by raphael ankierman on 22/05/2018.
+//  Copyright © 2018 CocoaPods. All rights reserved.
+//
+
+import Foundation
+import Quick
+import ModernAVPlayer
+import Nimble
+
+final class WaitingNetworkStateSpecs: QuickSpec {
+    
+    private var state: WaitingNetworkState!
+    private var mockPlayer = MockCustomPlayer()
+    private var url: URL!
+    private var playerMedia = ConcretePlayerMedia(url: URL(string: "x")!, type: .clip)
+    private var mockReachability: MockReachability!
+    private lazy var tested = ConcretePlayerContext(player: self.mockPlayer)
+    
+    override func spec() {
+        beforeEach {
+            self.mockReachability = MockReachability()
+            self.url = URL(string: "foo")!
+            self.state = WaitingNetworkState(context: self.tested,
+                                             urlToReload: self.url,
+                                             shouldPlaying: true,
+                                             error: CustomError.itemFailedWhenLoading,
+                                             reachabilityService: self.mockReachability)
+            self.tested.state = self.state
+        }
+        
+        context("init") {
+            it("should start reachability service") {
+                
+                // ASSERT
+                expect(self.mockReachability.start_callCount).to(equal(1))
+            }
+            
+            it("should setup isTimedOut callback") {
```

Comment:
Im nitpicking here...
Test is Ok, be title is not exact IMHO. You are not testing isTimedOut callback setting (in fact, we don't care).
You are testing that when reachability service times out, your state change to Failed (and that is the test needed indeed)